### ISSUE TITLE
Dynamic BGP Group Name Handling

### ISSUE THREAD

**Issue by jrokeach** [Juniper/meshrr#19]
Currently, 3 dynamically modified BGP groups are supported: MESHRR-MESH, MESHRR-CLIENTS, and MESHRR-UPSTREAM.

The goal of this enhancement is to support more than 3 groups by dynamically modifying allow lines. An example for this would be for cluster-external RR clients which will need to have different policies applied to them. Initially, 3 types of groups should be supported based on coredns:

1. Groups with dynamically generated `neighbor` statements (current `MESHRR-MESH` and `MESHRR-UPSTREAM` behavior)
2. Groups with dynamically generated `allow` statements consisting of all members of a group
3. Groups with dynamically generated `allow` statements consisting of all IPs within a defined set of ranges not in any other used group (current `MESHRR-CLIENTS` behavior)

This functionality should be based on configuration passed to the dynamic neighbor configuration sidecar and depends on #18. The execution functions on the sidecar should ingest group parameters into a standardized structure from coredns and be ready to be modified to ingest from other sources.

**Comment by jrokeach** [comment]
Initial thought for configuration structure:
```yaml
bgpgroups:
  - name: MESHRR-CLIENTS
    source: coredns
    output: neighbors
  - name: MESHRR-UPSTREAM
    source: coredns
    output: neighbors
  - name: MESHRR-CLIENTS
    source: coredns
    output: allow-subtractive
    ranges:
      - 172.16.0.0/12
      - 10.0.0.0/16
```

**Comment by jrokeach** [comment]
Implemented in v0.2 branch
